add-on/chrome/bootstrap.js

```diff
@@ -211,6 +211,11 @@ var WindowListener = {
             });
           };
 
+          if (PrivateBrowsingUtils.isWindowPrivate(window)) {
+            resolve(null);
+            return;
+          }
```

Comment:
This doesn't need to be inside the Promise callback. You can check immediately inside `openPanel` and `return Promise.resolve()` to short circuit.